test/itemlist_formaction.cpp

```diff
@@ -336,99 +340,207 @@ TEST_CASE("process_op(OP_TOGGLEITEMREAD)", "[itemlist_formaction]") {
 		REQUIRE_NOTHROW(itemlist.process_op(OP_TOGGLEITEMREAD));
 		REQUIRE(feed->unread_item_count() == 0);
 	}
+	SECTION("toggleitemread-jumps-to-next-unread") {
+		//NOTIMPL
+	}
 #endif
 }
 
 TEST_CASE("process_op(OP_SHOWURLS)", "[itemlist_formaction]") {
-	//NOTIMPL
+	INFO("Test");
+	controller c;
+	newsbeuter::view v(&c);
+	configcontainer cfg;
+	cache rsscache(":memory:", &cfg);
+	TestHelpers::TempFile urlFile;
+	std::string line;
+
+	std::string test_url = "http://test_url";
+	std::string test_title = "Article Title";
+	std::string test_author = "Article Author";
+	std::string test_description = "Article Description";
+	time_t test_pubDate = 42;
+	char test_pubDate_str[128];
+	strftime(test_pubDate_str, sizeof(test_pubDate_str), "%a, %d %b %Y %H:%M:%S %z", localtime(&test_pubDate));
+
+	std::string pager_prefix_title = "Title: ";
+	std::string pager_prefix_author = "Author: ";
+	std::string pager_prefix_date = "Date: ";
+	std::string pager_prefix_link = "Link: ";
+
+	v.set_config_container(&cfg);
+	c.set_view(&v);
+
+
+	std::shared_ptr<rss_feed> feed = std::make_shared<rss_feed>(&rsscache);
+	std::shared_ptr<rss_item> item = std::make_shared<rss_item>(&rsscache);
+
+	item->set_link(test_url);
+	item->set_title(test_title);
+	item->set_author(test_author);
+	item->set_description(test_description);
+	item->set_pubDate(test_pubDate);
+	itemlist_formaction itemlist(&v, itemlist_str);
+
+
+	SECTION("OP_SHOWURLS with external-url-viewer"){
+		feed->add_item(item);
+		itemlist.set_feed(feed);
+		cfg.set_configvalue("external-url-viewer", "tee > " + urlFile.getPath());
+	
+		REQUIRE_NOTHROW(itemlist.process_op(OP_SHOWURLS));
+
+		std::ifstream urlFileStream (urlFile.getPath());
+		REQUIRE( std::getline (urlFileStream,line) );
+		REQUIRE(line == pager_prefix_title + test_title);
+
+		REQUIRE( std::getline (urlFileStream,line) );
+		REQUIRE(line == pager_prefix_author + test_author);
+
+		REQUIRE( std::getline (urlFileStream,line) );
+		REQUIRE(line == pager_prefix_date + test_pubDate_str);
+
+		REQUIRE( std::getline (urlFileStream,line) );
+		REQUIRE(line == pager_prefix_link + test_url);
+
+		REQUIRE( std::getline (urlFileStream,line) );
+		REQUIRE(line == " ");
+
+		REQUIRE( std::getline (urlFileStream,line) );
+		REQUIRE(line == test_description);
+
+		REQUIRE( std::getline (urlFileStream,line) );
+		REQUIRE(line == "");
+	}
+
+	SECTION("OP_SHOWURLS default behaviour"){
+		feed->add_item(item);
+		itemlist.set_feed(feed);
+		REQUIRE_NOTHROW(itemlist.process_op(OP_SHOWURLS));
+	}
+
+	SECTION("OP_SHOWURLS empty feed"){
+		REQUIRE_NOTHROW(itemlist.process_op(OP_SHOWURLS));
```

Comment:
I totally agree, and as long as I find a return value or modified property I'll try to assert it. But on another hand some code branches will just check some value and abort, and I still think it is interesting to have the test class go through them both to ensure they don't crash and as a way to track dead code.
Just my two cents anyway, If you prefer I'll stick to testable results.